### PR TITLE
Fix build tags manipulation in Makefile

### PR DESCRIPTION
**Pull request by xueweiz**
This is for #366 

In the current Makefile, when ENABLE_JOURNAL is set to 0, BUILD_TAGS will be set to `""`. And result in below command:
```
CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build \
		-mod vendor \
		-o bin/node-problem-detector \
		-ldflags '-X k8s.io/node-problem-detector/pkg/version.version=v0.7.1-20-gc55fa33' \
		"" \
		./cmd/nodeproblemdetector
```

The `""` confuses golang into compiling the current directory :) which then causes the problem:
```
can't load package: package k8s.io/node-problem-detector: no Go files in /Users/mgrosser/Code/tools/node-problem-detector
make: *** [bin/node-problem-detector] Error 1
```

Also added a test case for the `ENABLE_JOURNAL=0` use case.

Also fixed a few rules for `log-counter` to deal with `ENABLE_JOURNAL=0`:
`log-counter` currently [only supports journald](https://github.com/kubernetes/node-problem-detector/blob/ef30a1fdd173198d1ce6475d6ed1f698759cee72/cmd/logcounter/log_counter.go#L42). So when `ENABLE_JOURNAL=0` is specified, the binary won't compile, and `docker build` won't find the binary to be installed into the docker image.

The fix basically skips the compilation and installation for `log-counter` when `ENABLE_JOURNAL=0`.
The change does not has any effect when `ENABLE_JOURNAL=1`.

### CODE REVIEW

**Comment by grosser** [comment]
thx for looking into this!

File: Makefile [hunk 1]
```diff
@@ -87,29 +84,38 @@ version:
...
+ifeq ($(ENABLE_JOURNALD), 1)
 	CGO_ENABLED=$(CGO_ENABLED) GOOS=linux GO111MODULE=on go build \
 		-mod vendor \
 		-o bin/log-counter \
 		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
-		$(BUILD_TAGS) \
+		-tags "$(BUILD_TAGS)" \
 		cmd/logcounter/log_counter.go
+else
+	echo "log-counter requires journald, skipping."
```
Comment (wangzhen127) [inline Makefile]: Nit: Print this as a warning:

echo "Warning: ..."

File: Makefile [hunk 2]
```diff
@@ -87,29 +84,38 @@ version:
...
 		-ldflags '-X $(PKG)/pkg/version.version=$(VERSION)' \
-		$(BUILD_TAGS) \
+		-tags "$(BUILD_TAGS)" \
 		./cmd/nodeproblemdetector
 
 Dockerfile: Dockerfile.in
 	sed -e 's|@BASEIMAGE@|$(BASEIMAGE)|g' $< >$@
+ifneq ($(ENABLE_JOURNALD), 1)
+	sed -i '/Below command depends on ENABLE_JOURNAL=1/,+2d' $@
+	echo "log-counter requires journald, skipping."
```
Comment (wangzhen127) [inline Makefile]: ditto

File: Makefile (hunk 1 above)
Comment (xueweiz) [reply Makefile]: Good point. Thanks!

File: Makefile (hunk 2 above)
Comment (xueweiz) [reply Makefile]: Thanks!

**Comment by xueweiz** [comment]
Thanks for the review Zhen! :)

**Pull request merged**